test/blackbox-tests/test-cases/preprocess-with-action-and-pps-errors.t

```diff
@@ -0,0 +1,42 @@
+Show an error when `(ppx ...)` appears in any position other than the last
```

Comment:
Might be useful to add a test for the case where `ppx` is used inside simple `rule`, e.g.

```
(rule
 (action
  (progn
   (run cat %{input-file})
   (ppx (some_ppx) %{input-file}.pp.1))))
```